pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1318,8 +1318,9 @@ public void deleteTopicAuthenticationWithRetry(String topic, CompletableFuture<V
     private CompletableFuture<Optional<Topic>> createNonPersistentTopic(String topic) {
         CompletableFuture<Optional<Topic>> topicFuture = new CompletableFuture<>();
         topicFuture.exceptionally(t -> {
+            // If the broker is not the owner of the topic, the client will redirect to another broker.
+            // In this case, we should not record as a topic load failure.
             pulsarStats.recordTopicLoadFailed();
-            pulsar.getExecutor().execute(() -> topics.remove(topic, topicFuture));
```

Comment:
Removing this line doesn't seem to be related to topic load failed metrics. Is this intentional?
It seems that in the original solution this would ensure that failed topics don't added to the topics map.